--- /data/data/com.termux/files/home/tmp/track-b-v10-v11-ultimate/sd_full/ggml/src/ggml-vulkan/ggml-vulkan.cpp	2026-08-26 10:14:50.713391100 +0900
+++ /data/data/com.termux/files/home/tmp/track-b-s21-v11-node-1055-resolution/src_sd/ggml/src/ggml-vulkan/ggml-vulkan.cpp	2026-08-26 16:24:54.789375219 +0900
@@ -9286,7 +9286,7 @@
     const ggml_type effective_src1_type = quantize_y ? GGML_TYPE_Q8_1 : (y_f32_kernel ? GGML_TYPE_F32 : src1->type);
 
     const uint32_t kpad = quantize_y ? 0 : ggml_vk_align_size(ne10, ggml_vk_guess_matmul_pipeline_align(ctx, mmp, ne01, ne11, qx_needs_dequant ? f16_type : src0->type, effective_src1_type));
-    const bool aligned = !quantize_y && ne10 == kpad && ne01 > 8 && ne11 > 8;
+    const bool aligned = !quantize_y && ne10 == kpad && (ne01 % 128 == 0) && (ne11 % 128 == 0);
 
     vk_pipeline pipeline = ggml_vk_guess_matmul_pipeline(ctx, mmp, ne01, ne11, aligned, qx_needs_dequant ? f16_type : src0->type, effective_src1_type);
 
@@ -10370,7 +10370,7 @@
     const ggml_type effective_src1_type = quantize_y ? GGML_TYPE_Q8_1 : (y_f32_kernel ? GGML_TYPE_F32 : src1->type);
 
     const uint32_t kpad = quantize_y ? 0 : ggml_vk_align_size(ne10, ggml_vk_guess_matmul_id_pipeline_align(ctx, mmp, ne01, nei1, qx_needs_dequant ? f16_type : src0->type, effective_src1_type));
-    const bool aligned = !quantize_y && ne10 == kpad && ne01 > 8 && nei1 > 8;
+    const bool aligned = !quantize_y && ne10 == kpad && (ne01 % 128 == 0) && (nei1 % 128 == 0);
 
     vk_pipeline pipeline = ggml_vk_guess_matmul_id_pipeline(ctx, mmp, ne01, nei1, aligned, qx_needs_dequant ? f16_type : src0->type, effective_src1_type);
 
